src/visualization/src/MeshcatVisualizer.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2023 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#include <MeshcatCpp/Material.h>
+#include <MeshcatCpp/Meshcat.h>
+#include <MeshcatCpp/Shape.h>
+
+#include <iDynTree/Core/Utils.h>
+#include <iDynTree/Core/MatrixFixSize.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/SolidShapes.h>
+#include <iDynTree/MeshcatVisualizer.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Model/ForwardKinematics.h>
+#include <iDynTree/Model/LinkState.h>
+#include <iDynTree/Model/Traversal.h>
+
+#include <memory>
+#include <string_view>
+#include <utility>
+
+using namespace iDynTree;
+
+struct MeshcatVisualizer::Impl
+{
+    ::MeshcatCpp::Meshcat meshcat;
+
+    struct ModelData
+    {
+        iDynTree::Model model;
+        iDynTree::Traversal traversal;
+        iDynTree::LinkPositions linkPositions;
+    };
+
+    std::unordered_map<std::string, ModelData> storedModels;
+    [[nodiscard]] inline bool modelExists(const std::string &modelName)
+    {
+        return this->storedModels.find(modelName) != this->storedModels.end();
+    }
+
+    [[nodiscard]] inline bool isMesh(const iDynTree::SolidShape &geometry)
+    {
+        if (!geometry.isExternalMesh())
+        {
+            return false;
+        }
+
+        std::string_view meshPath = geometry.asExternalMesh()->getFilename();
+
+        if (meshPath.size() == 0)
+        {
+            return false;
+        }
+
+        std::size_t pos = meshPath.find_last_of(".");
+        if (pos == std::string::npos)
+        {
+            return false;
+        }
+        const std::string_view format = meshPath.substr(pos + 1);
+        return format == "dae" || format == "stl" || format == "obj";
+    }
+
+    std::string getAbsoluteMeshPath(const std::string &modelName, const std::string &linkName, const iDynTree::ExternalMesh &externalMesh)
```

Comment:
Can you add a comment to clarify what this function is doing? From the name I tought it was returning a path in the filesystem, while by reading it I guess it is returning a path in the meshcat tree.